public/app/core/components/NavBar/Next/NavBarNext.tsx

```diff
@@ -55,8 +57,16 @@ export const NavBarNext = React.memo(() => {
   );
   const activeItem = isSearchActive(location) ? searchItem : getActiveItem(navTree, location.pathname);
   const [menuOpen, setMenuOpen] = useState(false);
+  const [queryParams] = useQueryParams();
+  const { path } = useRouteMatch();
+  const hideNavBar = getAppRoutes().some(
+    (r) =>
+      matchPath(path, { path: r.path, exact: r.exact }) &&
+      ((typeof r.hideNavBar === 'boolean' && r.hideNavBar) ||
+        (typeof r.hideNavBar === 'function' && r.hideNavBar(queryParams)))
+  );
```

Comment:
this looks messy, and a bit brittle maybe?

Not sure I like we are calling getAppRoutes on all re-renders of this component. 

you should be able to just get this from the routes form the current route?

Reply:
the function useRouteMatch , returns the route and it should have the hideNavBar property?

Reply:
`useRouteMatch` only returns a match of the form
```ts
{
  isExact: boolean;
  params: Record<string, string>;
  path: string;
  url: string;
}

Reply:
no, seems like you dont get the custom properties/route object

Reply:
if possible, a z-index fix seems easier.

Reply:
if possible, a z-index fix seems cleaner.

Reply:
not sure what you mean by z-index & FocusScope

Reply:
Lowering the z-index of the nav bar toggle so it doesn't appear on top of the dashboard settings, and FocusScope to trap keyboard navigation to dashboard settings when it's open